### PR TITLE
Fixed disable button enabled when url field is empty. 

### LINKED ISSUE #119: Disable save button when the URL field is empty
## Expected Behavior

If the URL field is empty the save button should be disabled

## Actual Behavior

It is possible to save a request with empty URL

## Steps to Reproduce the Problem

  1. Open Resting
  1. Leave the URL field empty
  1. Click save button

## Specifications

  - Addon Version:  All
  - Firefox Version: All
  - OS: All

### PR DESCRIPTION
**Pull request by vyasriday**
Fixes #119. Let me know if any changes required.

### CODE REVIEW

**Review by mirkoperillo — CHANGES_REQUESTED** [review]
Hi @vyasriday 
First of all thank you for investing time into Resting.Using jquery is a possible approach, by the way Resting is implemented using knockout.js so this way is preferred. I have wrote some hints to solve the problem using knockout.js
Feel free to ask for every doubtoach, by the way Resting is implemented using knockout.js so
this way is preferred.

Following some hints to solve the problem using knockout.js
Feel free to ask for every doubt

File: src/js/app.js [hunk 1]
```diff
@@ -775,7 +775,12 @@ requirejs(['jquery','app/storage','knockout','knockout-secure-binding','hjls','a
       } 
       Resting.tabContexts.remove(tab);
     };
-    
+
+   const disableSaveButton = (value) => {
```
Comment (mirkoperillo) [inline src/js/app.js:779]: Is this method used ?

File: src/js/app.js [hunk 2]
```diff
@@ -897,4 +902,13 @@ requirejs(['jquery','app/storage','knockout','knockout-secure-binding','hjls','a
   // appVM.communicationDialog();
   appVM.loadContexts();
   });
+  
+  $('#url-input-field').on('input', function (event) {
```
Comment (mirkoperillo) [inline src/js/app.js:906]: You can access the url field value using `Resting.request.url()`

File: src/index.html [hunk 3]
```diff
@@ -118,7 +118,7 @@
             <button type="button" class="btn btn-primary" data-bind="click: send">Send</button>
           </div>
           <div class="btn-group">
-            <button type="button" class="btn btn-primary" aria-haspopup="true" aria-expanded="false" data-bind="click: saveBookmarkDialog">Save</button>
+            <button type="button" id="save-button" class="btn btn-primary" aria-haspopup="true" aria-expanded="false" data-bind="click: saveBookmarkDialog" disabled="true">Save</button>
```
Comment (mirkoperillo) [inline src/index.html:121]: to enable or disable button you can use the `enable` binding into `data-bind` attribute.

More info how to use it [here](https://knockoutjs.com/documentation/enable-binding.html)

**Review by mirkoperillo — CHANGES_REQUESTED** [review]

File: src/js/app.js (hunk 1 above)
Comment (vyasriday) [reply src/js/app.js:779]: @mirkoperillo No it's not used. I will remove it.

**Comment by vyasriday** [comment]
@mirkoperillo Thanks for the review. I will try to make it work in knockoutjs once i get time.

**Comment by vyasriday** [comment]
@mirkoperillo Updated the PR with required changes. Let me know if anything else required.

**Comment by mirkoperillo** [comment]
@vyasriday there is a strange behavior: the button is enabled/disabled only when the focus is moved away from the url input field.

**Comment by vyasriday** [comment]
Yeah i saw that @mirkoperillo. I looked at the docs and this is how I found in docs. I though this is something related to knockout. May be it's not always listening for input change event. Please check code if there is anything I am missing.

File: src/js/app.js [hunk 4]
```diff
@@ -718,6 +718,8 @@ requirejs(['jquery','app/storage','knockout','knockout-secure-binding','hjls','a
       _activateTab(tabActivated);
     };
 
+    const enableSend = () => true
```
Comment (mirkoperillo) [inline src/js/app.js:721]: is this used ?

**Review by mirkoperillo — COMMENTED** [review]
Hi @vyasriday first at all thank you for contributing to Resting.
I found only an unused method, I think.
For the rest it seems good.
I'll give a deeper look into this problem when I will have time.
For the moment I cannot merge it, because the side effect kills the user experience, but the contribution is good

File: src/js/app.js (hunk 4 above)
Comment (vyasriday) [reply src/js/app.js:721]: I am sorry, I was into other things that I totally forgot it. I will remove it right now, and once I have time I will also try to understand why it's behaving this way. Meanwhile, if you come across any resources that can help me to better understand, please share it.

**Comment by vyasriday** [comment]
@mirkoperillo I will also try to understand why it's behaving this way. Meanwhile, if you come across any resources that can help me to better understand, please share it.

**Pull request merged**